src/ccache/ccache.cpp

```diff
@@ -2284,10 +2289,29 @@ hash_profiling_related_data(const Context& ctx, Hash& hash)
     // For a relative profile directory D the compiler stores $PWD/D as part of
     // the profile filename so we need to include the same information in the
     // hash.
-    const fs::path profile_path =
-      ctx.args_info.profile_path.is_absolute()
-        ? ctx.args_info.profile_path
-        : ctx.apparent_cwd / ctx.args_info.profile_path;
+    fs::path profile_path = ctx.args_info.profile_path.is_absolute()
+                              ? ctx.args_info.profile_path
+                              : ctx.apparent_cwd / ctx.args_info.profile_path;
+
+    if (!ctx.args_info.coverage_compilation_dir.empty()) {
+      profile_path = ctx.args_info.coverage_compilation_dir;
+    } else if (!ctx.args_info.coverage_prefix_map.empty()) {
+      for (const auto& map : ctx.args_info.coverage_prefix_map) {
```

Comment:
Should be possible to extract this loop to a common helper function ("apply_prefix_remapping") shared with the `-fdebug-prefix-map` case (logging can be done after the loop). I can do that later unless you want to do it as part of this PR.

Reply:
In practice I guess it makes no difference performance-wise since there will be at most a handful entries, but it sounds good to iterate backwards and stop after a match.